posthog/settings.py

```diff
@@ -123,19 +128,31 @@ def print_warning(warning_lines: Sequence[str]):
     "social_django",
 ]
 
-MIDDLEWARE = [
-    "posthog.middleware.SameSiteSessionMiddleware",  # keep this at the top
-    "django.middleware.security.SecurityMiddleware",
-    "posthog.middleware.AllowIP",
-    "django.contrib.sessions.middleware.SessionMiddleware",
-    "corsheaders.middleware.CorsMiddleware",
-    "django.middleware.common.CommonMiddleware",
-    "django.middleware.csrf.CsrfViewMiddleware",
-    "django.contrib.auth.middleware.AuthenticationMiddleware",
-    "django.contrib.messages.middleware.MessageMiddleware",
-    "django.middleware.clickjacking.XFrameOptionsMiddleware",
-    "whitenoise.middleware.WhiteNoiseMiddleware",
-]
+if STATSD_HOST:
+    MIDDLEWARE = [
+        "django_statsd.middleware.StatsdMiddleware",
+    ]
+else:
+    MIDDLEWARE = []
+
+MIDDLEWARE.extend(
+    [
+        "posthog.middleware.SameSiteSessionMiddleware",  # keep this at the top
+        "django.middleware.security.SecurityMiddleware",
+        "posthog.middleware.AllowIP",
+        "django.contrib.sessions.middleware.SessionMiddleware",
+        "corsheaders.middleware.CorsMiddleware",
+        "django.middleware.common.CommonMiddleware",
+        "django.middleware.csrf.CsrfViewMiddleware",
+        "django.contrib.auth.middleware.AuthenticationMiddleware",
+        "django.contrib.messages.middleware.MessageMiddleware",
+        "django.middleware.clickjacking.XFrameOptionsMiddleware",
+        "whitenoise.middleware.WhiteNoiseMiddleware",
+    ]
+)
+
+if STATSD_HOST:
+    MIDDLEWARE.append("django_statsd.middleware.StatsdMiddlewareTimer")
```

Comment:
We do this to make sure the first and last middlewares are statsd which the instructions suggest is a requirement.

Reply:
Might be cleaner instead to do `MIDDLEWARE.insert(0, "django_statsd.middleware.StatsdMiddleware")` and `MIDDLEWARE.append("django_statsd.middleware.StatsdMiddlewareTimer")`